packages/core/src/system.ts

```diff
@@ -1,13 +1,25 @@
-import { ActorSystem, ActorSystemInfo, AnyActorRef } from './types.js';
+import {
+  ActorSystem,
+  ActorSystemInfo,
+  AnyActorRef,
+  InspectionEvent,
+  Observer,
+  ResolvedInspectionEvent
+} from './types.ts';
+import { uniqueId } from './utils.ts';
 
-export function createSystem<T extends ActorSystemInfo>(): ActorSystem<T> {
-  let sessionIdCounter = 0;
+let systemCounter = 0;
+export function createSystem<T extends ActorSystemInfo>(
+  rootActor: AnyActorRef
+): ActorSystem<T> {
   const children = new Map<string, AnyActorRef>();
   const keyedActors = new Map<keyof T['actors'], AnyActorRef | undefined>();
   const reverseKeyedActors = new WeakMap<AnyActorRef, keyof T['actors']>();
+  const observers = new Set<Observer<ResolvedInspectionEvent>>();
 
   const system: ActorSystem<T> = {
-    _bookId: () => `x:${sessionIdCounter++}`,
+    root: rootActor,
+    _bookId: (name: string = 'x') => `${name}:${systemCounter++}`,
```

Comment:
This argument is never used in the codebase
```suggestion
    _bookId: () => `x:${systemCounter++}`,
```